src/Services.jl

```diff
@@ -1385,11 +1385,19 @@ gamelift(operation, args=[]) =
 gamelift(a...; b...) = gamelift(a..., b)
 
 function glacier(aws::AWSConfig, verb, resource, args=[])
+    version = "2012-06-01"
+    glacier_version_header = ["x-amz-glacier-version"=>version]
+
+    if args isa Pair && args.first == "headers"
+        append!(args[2], glacier_version_header)
+    else
+        args = vcat(args, ("headers" => glacier_version_header))
```

Comment:
What if `args` is a dict? Or when it's a list-like collection, doesn't this drop any original version of "headers"?

Reply:
Tests using all sorts of ways of passing `args` would be helpful in verifying correct behaviour.